py3status/modules/getjson.py

```diff
@@ -68,28 +65,17 @@ class Py3status:
     def getjson(self):
         """
         """
-        response = {
-            'cached_until': self.py3.time_in(self.cache_timeout),
-        }
-
         try:
-            resp = self.py3.request(self.url, timeout=self.timeout)
-            status = resp.status_code == 200
-            resp = resp.json()
+            getjson = self.py3.request(self.url, timeout=self.timeout).json()
+            getjson = self.py3.flatten_dict(
+                getjson, delimiter=self.delimiter, intermediates=True)
         except self.py3.RequestException:
-            resp = None
-            status = False
-
-        if status:
-            response['full_text'] = self.py3.safe_format(
-                self.format,
-                self.py3.flatten_dict(
-                    resp, delimiter=self.delimiter, intermediates=True
-                )
-            )
-        else:
-            response['full_text'] = ''
-        return response
+            getjson = None
```

Comment:
we'd just do this
```
return {
    'cached_until': self.py3.time_in(self.cache_timeout),
    'full_text': self.py3.safe_format(self.format, getjson),
}
```

The idea is less code everywhere.

So really the question is all about behaviour of the formatter.  I think we need to keep how this module works close to the original.

Yeah we'd need to see how other modules are affected.

So you might want to just do `full_text = ''` for a quick merge and then look at that issue over a few weeks or so.